src/uu/stty/src/stty.rs

```diff
@@ -462,6 +496,68 @@ fn apply_baud_rate_flag(termios: &mut Termios, input: &str) -> ControlFlow<bool>
     ControlFlow::Continue(())
 }
 
+fn apply_char_mapping(
+    termios: &mut Termios,
+    control_char_index: SpecialCharacterIndices,
+    new_val: &str,
+) -> Result<(), ControlCharMappingError> {
+    match string_to_control_char(new_val) {
+        Ok(val) => {
+            termios.control_chars[control_char_index as usize] = val;
+            Ok(())
+        }
+        Err(e) => Err(e),
+    }
+}
+
+// GNU stty defines some valid values for the control character mappings
+// 1. Standard character, can be a a single char (ie 'C') or hat notation (ie '^C')
+// 2. Integer
+//      a. hexadecimal, prefixed by '0x'
+//      b. octal, prefixed by '0'
+//      c. decimal, no prefix
+// 3. Disabling the control character: '^-' or 'undef'
+//
+// This function returns the ascii value of valid control chars, or None if the input is invalid
+fn string_to_control_char(s: &str) -> Result<u8, ControlCharMappingError> {
+    if s == "undef" || s == "^-" {
+        return Ok(0);
+    }
+
+    // try to parse integer (hex, octal, or decimal)
+    let mut ascii_num: Option<u32> = None;
+    if let Some(hex) = s.strip_prefix("0x") {
+        ascii_num = u32::from_str_radix(hex, 16).ok();
+    } else if let Some(octal) = s.strip_prefix("0") {
+        ascii_num = u32::from_str_radix(octal, 8).ok();
+    } else if let Ok(decimal) = s.parse::<u32>() {
+        ascii_num = Some(decimal);
+    }
```

Comment:
`ascii_num` doesn't have to be mutable. And you can get rid of the last `if let`:
```suggestion
    let ascii_num = if let Some(hex) = s.strip_prefix("0x") {
        u32::from_str_radix(hex, 16).ok()
    } else if let Some(octal) = s.strip_prefix("0") {
        u32::from_str_radix(octal, 8).ok()
    } else {
        s.parse::<u32>().ok()
    };
```